readthedocs/notifications/signals.py

```diff
@@ -43,3 +46,23 @@ def organization_disabled(instance, *args, **kwargs):
             message_id=MESSAGE_ORGANIZATION_DISABLED,
             attached_to=instance,
         )
+
+
+@receiver(post_save, sender=EmailAddress)
+def user_email_verified(instance, *args, **kwargs):
+    """Check if the primary email is validated and cancel the notification."""
+    if instance.primary:
+        if instance.verified:
+            Notification.objects.cancel(
+                attached_to=instance.user,
+                message_id=MESSAGE_EMAIL_VALIDATION_PENDING,
+            )
+        else:
+            Notification.objects.add(
```

Comment:
This is called every time a Email, Organization, Project are saved. It seems we're depending on `add` to not keep adding multiple notifications, and de-dupe things smartly. Is it safe to make that assumption? In particular, if they dismiss it, we won't try to re-add it right?

Reply:
Yes, that assumption is correct. That's how `.add()` should work. If it's not working like that, it's a bug and we should fix it.

If they dismiss it, we _will add a new one_, yes. We won't add a new one _only_ if there is already one in `READ` or `UNREAD` state only: https://github.com/readthedocs/readthedocs.org/blob/444bd6ff8f86f53e2be21b611686afd1d7b48285/readthedocs/notifications/querysets.py#L26-L43